### ISSUE TITLE
acc disappears from the result table

### ISSUE THREAD

**Issue by JiayuanSternLi** [ray-project/ray#6383]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.8.0dev
- **Python version**: 3.6
- **Exact command to reproduce**:

### Describe the problem

Acc disappears from the last column in the table. I  worked on the example code and added some helper functions as well as modifying some variables. However, the last column disappears and I don't know what modification caused.
For now, I could only print the stats dictionary 

### Source code / logs

`

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import argparse
import numpy as np
from tensorflow.keras.layers import Dense, Flatten, Conv2D
from tensorflow.keras import Model
from tensorflow.keras.datasets.fashion_mnist import load_data
from ray.tune.schedulers import AsyncHyperBandScheduler
from ray.tune.suggest.bayesopt import BayesOptSearch
import tensorflow_datasets as tfds
import read_params
from train_config import configure_model, configure_optimizer, configure_lossfunc

import ray
from ray import tune

FROTN_END_CALL = False

SPLIT_WEIGHTS = (8,1,1)
SPLITS = tfds.Split.TRAIN.subsplit(weighted=SPLIT_WEIGHTS)
IMG_SIZE = 224
IMG_SHAPE = (IMG_SIZE, IMG_SIZE, 3)
SHUFFLE_BUFFER_SIZE = 1000
BATCH_SIZE = 10
DATASET_NAME = 'tf_flowers'

class MyModel(Model):
    def __init__(self, hiddens=128):
        super(MyModel, self).__init__()
        import tensorflow as tf 
        self.conv1 = Conv2D(32, 3, activation="relu")
        self.flatten = Flatten()
        self.d1 = Dense(hiddens, activation="relu")
        self.d2 = Dense(10, activation="softmax")

    def call(self, x):
        x = self.conv1(x)
        x = self.flatten(x)
        x = self.d1(x)
        return self.d2(x)

def resize_normalize(image, label):
    import tensorflow as tf 
    image = tf.cast(image, tf.float32)
    if image.shape[-1] == 1:
        image = tf.concat([image, image, image], 2)
    image = tf.image.resize(image, method='mitchellcubic', size=(IMG_SIZE, IMG_SIZE))
    image = image / 255.0
    return image, label

def augment(image, label):
    import tensorflow as tf 
    image = tf.image.random_flip_left_right(image)
    image = tf.image.random_contrast(image, lower=0.0, upper=1.0)
    return image, label

def get_and_split_dataset(dataset_name):
    import tensorflow as tf 
    (raw_train, raw_val, raw_test), info = tfds.load(dataset_name,
                                                        split=list(SPLITS),
                                                        as_supervised=True,
                                                        with_info=True)
    return (raw_train, raw_val, raw_test), info

def get_dataset_stats(info):
    import tensorflow as tf 
    #---data info---#
    datasets_stats = {}
    train_num, val_num, test_num = (
      info.splits['train'].num_examples * weight/10 for weight in SPLIT_WEIGHTS
    )
    num_classes = info.features['label'].num_classes
    datasets_stats['train_num'] = train_num
    datasets_stats['val_num'] = val_num
    datasets_stats['test_num'] = test_num
    datasets_stats['classes_num'] = num_classes
    print(datasets_stats)
    return datasets_stats

def preprocess_data(datasets):
    import tensorflow as tf 
    (raw_train, raw_val, raw_test), info = datasets
    train = raw_train.map(resize_normalize)
    val = raw_val.map(resize_normalize)
    test = raw_test.map(resize_normalize)

    #---data augmentation---#
    # train = train.map(augment)

    #---data structure config----#
    shuffle_size = info.splits['train'].num_examples
    # train = train.shuffle(shuffle_size)
    train = train.batch(BATCH_SIZE)
    val = val.batch(BATCH_SIZE)
    test = test.batch(BATCH_SIZE)
    train = train.prefetch(tf.data.experimental.AUTOTUNE)
    return train, val, test

def assemble_model(num_classes, model_name='MobileNetV2'):
    import tensorflow as tf 
    base_model = tf.keras.applications.MobileNetV2(input_shape=IMG_SHAPE,
                                                    weights='imagenet',
                                                    include_top=True)
    model = tf.keras.Sequential([
                                base_model,
                                tf.keras.layers.Dense(500, activation='relu'),
                                tf.keras.layers.Dense(num_classes, activation='softmax')
                                ])
    return model

class Trainable(tune.Trainable):
    def _setup(self, config):
        # IMPORTANT: See the above note.
        import tensorflow as tf

        datasets, info = get_and_split_dataset(DATASET_NAME)
        print(info)
        self.train_ds, self.val_ds, self.test_ds = preprocess_data((datasets, info))
        self.num_classes = info.features['label'].num_classes

        # self.model = tf.keras.applications.MobileNetV2( weights='imagenet')
        # self.model = assemble_model(self.num_classes, model_name='MobileNetV2')     
        self.model = MyModel()
        

        self.optimizer = configure_optimizer(config)
        
        # self.optimizer = configure_optimizer(config)

        self.loss_object = tf.keras.losses.SparseCategoricalCrossentropy()
        self.optimizer.learning_rate = config.get("lr", 0.01)
        # self.optimizer.momentum = config.get("momentum", 0)
        # self.optimizer.nesterov = config.get("nesterov", False)
        self.train_loss = tf.keras.metrics.Mean(name="train_loss")
        self.train_accuracy = tf.keras.metrics.SparseCategoricalAccuracy(
            name="train_accuracy")

        self.val_accuracy = tf.keras.metrics.SparseCategoricalAccuracy(
            name="val_accuracy")
        self.test_loss = tf.keras.metrics.Mean(name="test_loss")
        self.test_accuracy = tf.keras.metrics.SparseCategoricalAccuracy(
            name="test_accuracy")

        @tf.function
        def train_step(images, labels):
            with tf.GradientTape() as tape:
                predictions = self.model(images)
                loss = self.loss_object(labels, predictions)
            gradients = tape.gradient(loss, self.model.trainable_variables)
            self.optimizer.apply_gradients(
                zip(gradients, self.model.trainable_variables))

            self.train_loss(loss)
            self.train_accuracy(labels, predictions)

        @tf.function 
        def val_step(images, labels):
            predictions = self.model(images)
            self.val_accuracy(labels, predictions)

        @tf.function
        def test_step(images, labels):
            predictions = self.model(images)
            # labels = tf.expand_dims(labels, -1)
            # t_loss = self.loss_object(labels, predictions)
            # self.test_loss(t_loss)
            self.test_accuracy(labels, predictions)

        self.tf_train_step = train_step
        # self.tf_val_step = val_step
        self.tf_test_step = test_step

    def _train(self):
        self.train_loss.reset_states()
        self.train_accuracy.reset_states()
        # self.test_loss.reset_states()
        self.test_accuracy.reset_states()
        self.val_accuracy.reset_states()

        for idx, (images, labels) in enumerate(self.train_ds):
            self.tf_train_step(images, labels)

        # for val_images, val_labels in self.val_ds:
        #     self.tf_val_step(val_images, val_labels)

        for test_images, test_labels in self.test_ds:
            self.tf_test_step(test_images, test_labels)

        # It is important to return tf.Tensors as numpy objects.
        print({
            "epoch": self.iteration,
            "train_loss": self.train_loss.result().numpy(),
            "train_accuracy": self.train_accuracy.result().numpy() * 100,
            # "val_accuracy": self.val_accuracy.result().numpy() * 100,
            "test_accuracy": self.test_accuracy.result().numpy() * 100
        })
        return {
            "epoch": self.iteration,
            "train_loss": self.train_loss.result().numpy(),
            "train_accuracy": self.train_accuracy.result().numpy() * 100,
            # "val_accuracy": self.val_accuracy.result().numpy() * 100,
            "test_accuracy": self.test_accuracy.result().numpy() * 100
        }

def get_config(discrt_pm, cont_pm):
    config = {}
    for key, val in discrt_pm.items():
        config[key] = tune.choice(val)
    
    print(config)
    for key, val in cont_pm.items():
        config[key] = tune.uniform(val[0], val[1])
    print(config)

    return config

if __name__ == "__main__":
    import tensorflow as tf 
    ray.shutdown()
    ray.init()
    scheduler = AsyncHyperBandScheduler(time_attr='training_iteration', 
                                        metric="acc", 
                                        mode="max")

    continue_params, discrete_params, notuning_params  = read_params.get_param_dict()
    config = get_config(discrete_params, continue_params)
    print("continue_params:", continue_params)
    print("discrete_params:", discrete_params)
    print("notuning_params", notuning_params)

    print("config", config)
    analysis = tune.run(
        Trainable,
        stop={"training_iteration": 5},
        verbose=1,
        num_samples=3,
        scheduler=scheduler,
        config=config,
               
        # config= {
        #        # "hiddens": tune.choice([32, 64, 128]),
        #        "lr": tune.sample_from(lambda spec: np.random.uniform(0.05, 0.1)),
        #        "optimizer": tune.choice(["sgd", "adam"])
        #        },
        resources_per_trial={'gpu': 4, 'cpu': 40},
        name = "example-experiment"
        )

    print("Best config is", analysis.get_best_config(metric="train_accuracy"))
    trials = analysis.trials
    print("All trials inform:", trials)

`
![image](https://user-images.githubusercontent.com/33815430/70323611-3995b780-1868-11ea-9db7-0d88150f92f7.png)

**Comment by ericl** [comment]
cc @ujvl

**Comment by ujvl** [comment]
fields that aren't populated are hidden from the table to save space

@JiayuanSternLi you need to return `mean_accuracy` in the result object

**Comment by ujvl** [comment]
Feel free to reopen if you see further issues.

**Comment by JiayuanSternLi** [comment]
> fields that aren't populated are hidden from the table to save space
> 
> @JiayuanSternLi you need to return `mean_accuracy` in the result object

Thank you, it works... It looks like test_accuracy must be named as "mean_accuracy" in return results.
